com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -88,6 +89,34 @@ public UnityRLInitParameters Initialize(CommunicatorInitParameters initParameter
                         RlInitializationOutput = academyParameters
                     },
                     out input);
+
+                // Initialization succeeded part-way. The most likely cause is a mismatch between the communicator
+                // API strings, so log an explicit warning if that's the case.
+                if (initializationInput != null && input == null)
```

Comment:
Note that previously, this would raise a NullReferenceException on the line when we access `input.RlInput`. This logs the warning and raises an explicit error instead. 
https://github.com/Unity-Technologies/ml-agents/blob/1bc602f7bd78d81417860ac19ee9f61f3cc9cc69/com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs#L108